src/pages/settings/Profile/Contacts/NewContactMethodPage.js

```diff
@@ -0,0 +1,159 @@
+import TextInput from '../../../../components/TextInput';
+import withLocalize, {withLocalizePropTypes} from '../../../../components/withLocalize';
+import Navigation from '../../../../libs/Navigation/Navigation';
+import Permissions from '../../../../libs/Permissions';
+import ONYXKEYS from '../../../../ONYXKEYS';
+import ROUTES from '../../../../ROUTES';
+import styles from '../../../../styles/styles';
+import * as User from '../../../../libs/actions/User';
+import * as LoginUtils from '../../../../libs/LoginUtils';
+
+const propTypes = {
+    /* Onyx Props */
+
+    /** List of betas available to current user */
+    betas: PropTypes.arrayOf(PropTypes.string),
+
+    /** Login list for the user that is signed in */
+    loginList: PropTypes.shape({
+        /** The partner creating the account. It depends on the source: website, mobile, integrations, ... */
+        partnerName: PropTypes.string,
+
+        /** Phone/Email associated with user */
+        partnerUserID: PropTypes.string,
+
+        /** The date when the login was validated, used to show the brickroad status */
+        validatedDate: PropTypes.string,
+
+        /** Field-specific server side errors keyed by microtime */
+        errorFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+
+        /** Field-specific pending states for offline UI status */
+        pendingFields: PropTypes.objectOf(PropTypes.objectOf(PropTypes.string)),
+    }),
+
+    ...withLocalizePropTypes,
+};
+const defaultProps = {
+    betas: [],
+    loginList: {},
+};
+
+class NewContactMethodPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            login: '',
+            password: '',
+        };
+        this.onLoginChange = this.onLoginChange.bind(this);
+        this.validateForm = this.validateForm.bind(this);
+        this.submitForm = this.submitForm.bind(this);
+    }
+
+    onLoginChange(login) {
+        this.setState({login});
+    }
+
+    /**
+    * Determine whether the form is valid
+    *
+    * @returns {Boolean}
+    */
+    validateForm() {
+        const login = this.state.login.trim();
+        const phoneLogin = LoginUtils.getPhoneNumberWithoutSpecialChars(login);
+
+        return (Permissions.canUsePasswordlessLogins(this.props.betas) || this.state.password)
+            && (Str.isValidEmail(login) || Str.isValidPhone(phoneLogin));
+    }
+
+    submitForm() {
+        // If this login already exists, just go back.
+        if (lodashGet(this.props.loginList, this.state.login, null)) {
+            Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHODS);
+            return;
+        }
+        User.addNewContactMethodAndNavigate(this.state.login, this.state.password);
+    }
+
+    render() {
+        return (
+            <ScreenWrapper
+                onTransitionEnd={() => {
+                    if (!this.loginInputRef) {
+                        return;
+                    }
```

Comment:
Is this possible? What are the cases where the ref would not yet be available, but the transition ended? :confused-dog:

Reply:
I don't know when exactly this is possible but I see that we always check if the input ref is null when we auto-focus at the end of the screen transition.

Reply:
Ah yes, well, sometimes the code is immortal. I've seen these before. Things in the code that nobody knows why we do them so they have life eternal. Nobody is brave enough to challenge them.